From a5f41872102b14c4382f307b0a2c656f437cd4ae Mon Sep 17 00:00:00 2001
From: Vaibhav Hiremath <hvaibhav@ti.com>
Date: Thu, 28 Jun 2012 15:19:44 +0800
Subject: [PATCH 011/609] arm:omap:am33xx: add control_status register addr &
 bit offsets

Original commit: 020b5897e6548e7ef18f7f362a1748f8ddafaa6e

  From git://arago-project.org/git/projects/linux-am33x.git
  And ti-sdk-am335x-evm-05.05.00.00-Linux-x86-Install image

This information is required for clock tree data,
where we need to read the control_status register
to detect the input Osc freq.
Here control_status register is used for .clksel_reg.

Signed-off-by: Vaibhav Hiremath <hvaibhav@ti.com>
---
 arch/arm/mach-omap2/control.h |   15 +++++++++++++++
 1 files changed, 15 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-omap2/control.h b/arch/arm/mach-omap2/control.h
index a406fd0..35f863f 100644
--- a/arch/arm/mach-omap2/control.h
+++ b/arch/arm/mach-omap2/control.h
@@ -337,6 +337,21 @@
 #define AM35XX_HECC_SW_RST		BIT(3)
 #define AM35XX_VPFE_PCLK_SW_RST		BIT(4)
 
+/* AM33XX CONTROL_STATUS bits */
+#define AM33XX_SYSBOOT0			(0xff << 0)
+#define AM33XX_DEVTYPE			(1 << 8)
+#define AM33XX_GPMC_CS0_BW		(1 << 16)
+#define AM33XX_GPMC_CS0_WAITEN		(1 << 17)
+#define AM33XX_GPMC_CS0_ADMUX		(0x3 << 18)
+#define AM33XX_SYSBOOT1			(0x3 << 22)
+
+/*
+ * CONTROL AM33XX STATUS register to identify boot-time configurations
+ */
+#define AM33XX_CONTROL_STATUS_OFF	0x040
+#define AM33XX_CONTROL_STATUS		AM33XX_L4_WK_IO_ADDRESS(AM33XX_CTRL_BASE + \
+						AM33XX_CONTROL_STATUS_OFF)
+
 /*
  * CONTROL AM33XX STATUS register
  */
-- 
1.7.5.4

